features/nanostack/mbed-mesh-api/mbed-mesh-api/mesh_interface_types.h

```diff
@@ -64,6 +64,93 @@ typedef enum {
+    /* Buffers */
+    uint32_t buf_alloc;             /*<! Buffer allocation count. */
+    uint32_t buf_headroom_realloc;  /*<! Buffer headroom realloc count. */
+    uint32_t buf_headroom_shuffle;  /*<! Buffer headroom shuffle count. */
+    uint32_t buf_headroom_fail;     /*<! Buffer headroom failure count. */
+    /* ETX */
+    uint16_t etx_1st_parent;        /*<! Primary parent ETX. */
+    uint16_t etx_2nd_parent;        /*<! Secondary parent ETX. */
+    /* MAC */
+    uint16_t adapt_layer_tx_queue_size; /*<! Adaptation layer direct TX queue size. */
+    uint16_t adapt_layer_tx_queue_peak; /*<! Adaptation layer direct TX queue size peak. */
+    /* Wi-SUN */
+    uint32_t asynch_tx_count;  /*<! Asynch TX counter */
+    uint32_t asynch_rx_count;  /*<! Asynch RX counter */
+} mesh_nw_statistics_t;
+
+/**
+ * Mesh physical layer statistics
+ */
+typedef struct {
+    uint32_t crc_fails;    /*<! CRC failures */
+    uint32_t tx_timeouts;  /*<! transmission timeouts */
+    uint32_t rx_timeouts;  /*<! reception timeouts */
+} mesh_phy_statistics_t;
+
+/**
+ * Mesh MAC statistics
+ */
+typedef struct {
+    uint16_t mac_tx_queue_size;       /*<! MAC TX queue current size. */
+    uint16_t mac_tx_queue_peak;       /*<! MAC TX queue peak size. */
+    uint32_t mac_rx_count;            /*<! MAC RX packet count. */
+    uint32_t mac_tx_count;            /*<! MAC TX packet count. */
+    uint32_t mac_bc_rx_count;         /*<! MAC broadcast RX packet count. */
+    uint32_t mac_bc_tx_count;         /*<! MAC broadcast TX packet count. */
+    uint32_t mac_beacon_rx_count;     /*<! MAC Beacon RX packet count. */
+    uint32_t mac_beacon_tx_count;     /*<! MAC Beacon TX packet count. */
+    uint32_t mac_rx_drop_count;       /*<! MAC RX packet drop count. */
+    uint32_t mac_tx_bytes;            /*<! MAC TX bytes count. */
+    uint32_t mac_rx_bytes;            /*<! MAC RX bytes count. */
+    uint32_t mac_tx_failed_count;     /*<! MAC TX failed count. */
+    uint32_t mac_retry_count;         /*<! MAC TX retry count. */
+    uint32_t mac_cca_attempts_count;  /*<! MAC CCA attempts count. */
+    uint32_t mac_failed_cca_count;    /*<! MAC failed CCA count. */
+} mesh_mac_statistics_t;
+
+/**
+ * Mesh frequency hopper statistics
+ */
+typedef struct {
+    int16_t fhss_drift_compensation;  /*<! FHSS synchronization drift compensation (us/channel). */
+    uint8_t fhss_hop_count;           /*<! FHSS hop count. */
+    uint16_t fhss_synch_interval;     /*<! FHSS synchronization interval (s). */
+    int16_t fhss_prev_avg_synch_fix;  /*<! Average of 5 preceding synchronization fixes (us). Updated after every fifth synch fix. */
+    uint32_t fhss_synch_lost;         /*<! FHSS synchronization lost counter. */
+    uint32_t fhss_unknown_neighbor;   /*<! FHSS TX to unknown neighbour counter. */
+    uint32_t fhss_channel_retry;      /*<! FHSS channel retry counter. */
+} mesh_fhss_statistics_t;
+
```

Comment:
Take everything ever defined is not The best approach 

We need to have some thought what are the needed and usable statistics this will make the messages huge

Reply:
removed not needed statistics